### DISCUSSION TITLE
'run-after-fail' does not send request, if the backup can not be created due to resticprofile process still running

### DISCUSSION THREAD

**Discussion by fabianzimmmermann** [creativeprojects/resticprofile#419] [Q&A] [answered]
We're using the `run-after-fail `configuration to get notified about failed backups. However, there are two situations where backups aren't being created:

1. The backup process exceeds the interval time between backups, causing the process to keep running without creating a backup.
2. Someone manually mounts the snapshots to a mount point and migrates data from the mounted volume. If the mount isn't unmounted afterward, the process keeps running.

In these scenarios, `run-after-fail` doesn't send any POST requests to our monitoring system. Is there a built-in method to receive these notifications, or do we need to implement our own solution?

Thanks

**Comment by creativeprojects** [comment, marked as answer]
Yes: it is a bit of a chicken and egg issue.

The lock always comes first, and all targets are running inside the lock.
This is by design: imagine you need to mount a network drive or something, if it's done outside the lock it might break another running task.

I was thinking about giving a choice in the configuration: either `run-after-fail` runs inside the lock, or outside, assuming the user understand the consequences.

There's a diagram showing the whole process: https://creativeprojects.github.io/resticprofile/configuration/run_hooks/index.html#order-of-run--during-a-backup

**Reply by fabianzimmmermann** [reply]
Thank you for the quick and clear answer!
We will create a job to test for these cases.
We love resticprofile!
